openmdao/components/cross_product_comp.py

```diff
@@ -0,0 +1,133 @@
+"""Definition of the Cross Product Component."""
+
+from six import string_types
+
+import numpy as np
+
+from openmdao.api import ExplicitComponent
+
+
+class CrossProductComp(ExplicitComponent):
+    """
+    Compute a vectorized dot product.
+
+    math::
+        c = np.cross(a, b)
+
+    where a is of shape (vec_size, 3)
+          b is of shape (vec_size, 3)
+          c is of shape (vec_size, 3)
+
+    if vec_size > 1 and
+
+    where a is of shape (3,)
+          b is of shape (3,)
+          c is of shape (3,)
+
+    otherwise.
+    """
+
+    def initialize(self):
+        """
+        Declare metadata.
+        """
+        self.metadata.declare('vec_size', types=int, default=1,
+                              desc='The number of points at which the dot product is computed')
+        self.metadata.declare('a_name', types=string_types, default='a',
+                              desc='The variable name for vector a.')
+        self.metadata.declare('b_name', types=string_types, default='b',
+                              desc='The variable name for vector b.')
+        self.metadata.declare('c_name', types=string_types, default='c',
+                              desc='The variable name for vector c.')
+        self.metadata.declare('a_units', types=string_types, default=None, allow_none=True,
```

Comment:
Many of these options (for example: a_units) spread across the three components are:

1. Not covered by tests.
2. Not detailed in feature docs.